dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Intrinsics.java

```diff
@@ -400,6 +403,84 @@ private List<Event> inlineAtomicEnd(FunctionCall ignored) {
         return List.of(EventFactory.Svcomp.newEndAtomic(checkNotNull(currentAtomicBegin)));
     }
 
+    private final static FunctionType PTHREAD_THREAD_TYPE = types.getFunctionType(
+            types.getPointerType(), List.of(types.getPointerType())
+    );
+
+    private List<Event> inlinePthreadCreate(FunctionCall call) {
+        final List<Expression> arguments = call.getArguments();
+        assert arguments.size() == 4;
+        final Expression pidResultAddress = arguments.get(0);
+        //final Expression attributes = arguments.get(1);
+        final Expression targetFunction = arguments.get(2);
+        final Expression argument = arguments.get(3);
+
+        final Register resultRegister = getResultRegister(call);
+        assert resultRegister.getType() instanceof IntegerType;
+
+        final Register tidReg = call.getFunction().newUniqueRegister("__tid", types.getArchType());
+        final DynamicThreadCreate createEvent = newDynamicThreadCreate(tidReg, PTHREAD_THREAD_TYPE, targetFunction, List.of(argument));
+
+        return eventSequence(
+                createEvent,
+                newStore(pidResultAddress, tidReg),
+                // TODO: Allow to return failure value (!= 0)
+                newLocal(resultRegister, expressions.makeZero((IntegerType) resultRegister.getType()))
+        );
+    }
+
+    private List<Event> inlinePthreadJoin(FunctionCall call) {
+        final List<Expression> arguments = call.getArguments();
+        assert arguments.size() == 2;
+        final Expression tidExpr = arguments.get(0);
+        final Expression returnAddr = arguments.get(1);
+        final boolean hasReturnAddr = !(returnAddr instanceof IntLiteral lit && lit.isZero());
+
+        final Register resultRegister = getResultRegister(call);
+        assert resultRegister.getType() instanceof IntegerType;
+
+        final Type joinType = types.getAggregateType(List.of(types.getIntegerType(8), PTHREAD_THREAD_TYPE.getReturnType()));
+        final Register joinReg = call.getFunction().newUniqueRegister("__joinReg", joinType);
+
+        final Expression status = expressions.makeExtract(joinReg, 0);
+        final Expression retVal = expressions.makeExtract(joinReg, 1);
+
+        final Expression statusSuccess = expressions.makeValue(SUCCESS.ordinal(), (IntegerType) status.getType());
```

Comment:
This means `SUCCESS` need always to be first in the enum, right? If so, this should be documented in the enum itself.